swig/include/python/python_exceptions.i

```diff
@@ -392,16 +395,19 @@ def DontUseExceptions():
     """ Disable exceptions in all GDAL related modules (osgeo.gdal, osgeo.ogr, osgeo.osr, osgeo.gnm).
         Note: prior to GDAL 3.7, this only affected the calling module"""
 
+    import importlib.util
+
     try:
         from . import gdal
         gdal._DontUseExceptions()
     except ImportError:
         pass
-    try:
-        from . import gdal_array
-        gdal_array._DontUseExceptions()
-    except ImportError:
-        pass
+    if importlib.util.find_spec("osgeo.gdal_array") and importlib.util.find_spec("numpy"):
```

Comment:
is it possible to wrap this code into a "private" function  to avoid duplication?

Reply:
fixed